### PR TITLE
Improved UX for permission list

### LINKED ISSUE #10747: UX: Permissions list
Would be nice to get user & team avatars in this list as well as user full name (only showing login/username currently).

The Icon is the only thing that separates Users from Teams. Maybe we need a label instead? Team: User: and then we can use the avatar img instead. 

screen shot 2018-02-03 at 09 58 13

thoughts @mattttt ? @Ijin08 can you do a mockup?

### PR DESCRIPTION
**Pull request by Ijin08**
Fixes #10747

skarmavbild 2018-03-28 kl 16 25 11

### CODE REVIEW

**Comment by torkelo** [comment]
this needs to be rebased on master (please have daniel or someone help you with that)

**Review by marefr — CHANGES_REQUESTED** [review]
Code looks good :+1: 

* When adding a user or team permission the avatar is missing. Refreshing the page and the avatar are rendered correctly.
* /pkg/api/folder_permission.go should also have the 
UserAvatarUrl and TeamAvatarUrl populated.
* Icons for admin, editor, viewer should be bigger. I barely see them now. Not exactly sure what was decided regarding which icons to use but would be nice with different icons for viewer and editor.
* Documentation of permissions probably needs updated image. Also http api for folder and dashboard permissions needs to be updated with UserAvatarUrl ,TeamAvatarUrl and TeamEmail. 

Docs can wait until this has been merged I suppose.

File: public/app/core/components/Permissions/DisabledPermissionsListItem.tsx
Comment (marefr) [inline public/app/core/components/Permissions/DisabledPermissionsListItem.tsx]: Would be nice with a specific gicon-class name for this in scss if not already exists. Seems like this style is needed as well in PermissionListItem.tsx (when there are no avatar).

File: public/app/core/components/Permissions/PermissionsListItem.tsx
Comment (marefr) [inline public/app/core/components/Permissions/PermissionsListItem.tsx]: Remove

File: public/app/stores/PermissionsStore/PermissionsStore.ts
Comment (marefr) [inline public/app/stores/PermissionsStore/PermissionsStore.ts]: Same is needed for setTeam. When you add a new team permission the team avatar is not rendered

**Review by marefr — CHANGES_REQUESTED** [review]
I pushed some backend changes so that correct team avatar is returned (default if no team email set).

You've fixed my earlier change request partially, i.e. works for user, but not for teams. Size of icons are applies to disabled rows, but not role permissions.

**Comment by Ijin08** [comment]
@marefr The viewer and editor icons aren't 100% decided yet. Will talk with Trent about them.

**Review by marefr — APPROVED** [review]
Nice! Approving this, but won't merge until the icons have been sorted out.

One question/concern if we should sort permessions or not. Seems to be sorted now by time they as added, see screenshot:

image

**Comment by Ijin08** [comment]
@marefr Yes they should be sorted: role-team-user. I guess the roles should be sorted admin-editor-viewer. And teams and roles alphabetically. 

@torkelo your thoughts on this?

**Comment by torkelo** [comment]
yes, should be sorted like you suggest.

**Comment by marefr** [comment]
@Ijin08 @torkelo I guess there's an edge case when having inherited permissions, or not? Feels natural for me that the inherited ones are displayed first and then the others. Both sorted according to suggested. Don't know if this makes sense, but something to discuss/think of.

**Comment by torkelo** [comment]
yea, inherited should be first, I agree

**Review by marefr — APPROVED** [review]
Great work :+1:

**Review by marefr — CHANGES_REQUESTED** [review]
Would like to have some tests of the PermissionStore (if possibly) to make it more clear what the expected sort order are.

**Review by marefr — APPROVED** [review]

**Comment by marefr** [comment]
Awesome work @Ijin08 :+1:

**Pull request merged**
